packages/template/__tests__/emit-component.test.ts

```diff
@@ -40,31 +41,27 @@ class MyComponent<T> extends TestComponent<MyComponentSignature<T>> {
    */
   static {
     templateForBackingValue(this, function (𝚪) {
-      const component = emitComponent(resolve(globals.let)({}, 𝚪.this.state.ready));
+      const component = emitComponent(resolve(globals.let)(𝚪.this.state.ready));
       const [isReady] = component.blockParams.default;
 
       {
         const 𝛄 = emitElement('div');
         expectTypeOf(𝛄).toEqualTypeOf<{ element: HTMLDivElement }>();
-        applyModifier(𝛄.element, resolve(globals.on)({}, 'click', 𝚪.this.wrapperClicked));
+        applyModifier(resolve(globals.on)(𝛄.element, 'click', 𝚪.this.wrapperClicked));
```

Comment:
Thinking about the general implementation, and looking at this specific example—to your point about the representation changing if the default modifier manager is approved and ships something *different* than GlimmerX ships, do we have any risk from that? Particularly, do we have any risk of stuff that will surface as a breaking change for our consumers?

Reply:
In principle this should be invisible to consumers (other than incidentally fixing the variance issue I mentioned, but we would've done that regardless). If Ember ships something different than GlimmerX, what we'll likely do is update the internal representation to match that and add a `resolve()` signature to the GlimmerX environment that converts its function signature to whatever Ember lands on.

While this _could_ have observable effects for GlimmerX users, the chances are slimmer there since modifiers tend not to do a lot of the type gymnastics that helpers do (no assertions/guards, no parametric return types, etc). I also imagine it's reasonably likely that GlimmerX would eventually align with Ember, allowing us to drop that extra resolution, but if not it's still not a huge deal.